src/analysis/inner_product_space/cross_product.lean

```diff
@@ -0,0 +1,181 @@
+  linear_equiv.coe_coe, linear_equiv.trans_apply, linear_isometry_equiv.coe_to_linear_equiv,
+  linear_isometry_equiv.norm_map, submodule.coe_norm,
+  inner_product_space.to_dual_symm_apply, alternating_map.curry_left_apply_apply,
+  alternating_map.const_linear_equiv_of_is_empty_symm_apply,
+  eq_self_iff_true, linear_map.coe_to_continuous_linear_map', matrix.zero_empty]
+
+/-- The cross-product is antisymmetric: `x ×₃ y = - (y ×₃ x)`. -/
+lemma cross_product_swap (x y : E) : x ×₃ y = - (y ×₃ x) :=
+begin
+  apply ext_inner_right ℝ,
+  intros z,
+  rw [inner_neg_left, o.inner_cross_product_apply, o.inner_cross_product_apply],
+  convert o.volume_form.map_swap ![y, x, z] (_ : (0 : fin 3) ≠ 1),
+  { ext i,
+    fin_cases i; refl },
+  { norm_num }
+end
+
+/-- The cross-product is antisymmetric: the cross-product of a vector with itself is 0. -/
+lemma cross_product_apply_self (v : E) : v ×₃ v = 0 :=
+begin
+  apply ext_inner_right ℝ,
+  intros w,
+  rw [inner_cross_product_apply, inner_zero_left],
+  refine o.volume_form.map_eq_zero_of_eq ![v, v, w] _ (_ : (0 : fin 3) ≠ 1),
+  { simp },
+  { norm_num }
+end
+
+/-- The cross-product of `u` and `v` is orthogonal to `u`. -/
+lemma inner_cross_product_apply_self (u v : E) : ⟪u ×₃ v, u⟫ = 0 :=
+begin
+  rw o.inner_cross_product_apply u v u,
+  refine o.volume_form.map_eq_zero_of_eq ![u, v, u] _ (_ : (0 : fin 3) ≠ 2),
+  { simp },
+  { norm_num }
+end
+
+/-- The cross-product of `u` and `v` is orthogonal to `v`. -/
+lemma inner_cross_product_apply_apply_self (u v : E) : ⟪u ×₃ v, v⟫ = 0 :=
+by rw [cross_product_swap, inner_neg_left, inner_cross_product_apply_self, neg_zero]
+
+/-- The map `cross_product`, upgraded from linear to continuous-linear; useful for calculus. -/
+def cross_product' : E →L[ℝ] (E →L[ℝ] E) :=
+(↑(linear_map.to_continuous_linear_map : (E →ₗ[ℝ] E) ≃ₗ[ℝ] (E →L[ℝ] E))
+  ∘ₗ (o.cross_product)).to_continuous_linear_map
+
+@[simp] lemma cross_product'_apply (v : E) :
+  o.cross_product' v = (o.cross_product v).to_continuous_linear_map :=
+rfl
+
+/-- The norm of the cross-product of `u` and `v` is bounded by the product of the norms of `u` and
+`v`. -/
+lemma norm_cross_product_le (u v : E) : ∥u ×₃ v∥ ≤ ∥u∥ * ∥v∥ :=
+begin
+  cases eq_or_lt_of_le (norm_nonneg (u ×₃ v)) with h h,
+  { rw ← h,
+    positivity },
+  refine le_of_mul_le_mul_right _ h,
+  rw ← real_inner_self_eq_norm_mul_norm,
+  simpa only [inner_cross_product_apply, fin.mk_zero, fin.prod_univ_succ, finset.card_singleton,
+    finset.prod_const, fintype.univ_of_subsingleton, matrix.cons_val_fin_one, matrix.cons_val_succ,
+    matrix.cons_val_zero, mul_assoc, nat.nat_zero_eq_zero, pow_one, submodule.coe_norm]
+    using o.volume_form_apply_le ![u, v, u ×₃ v]
+end
+
+/-- For orthogonal vectors `u` and `v`, the norm of the cross-product of `u` and `v` is the product
+of the norms of `u` and `v`. -/
+lemma norm_cross_product (u : E) (v : (ℝ ∙ u)ᗮ) : ∥u ×₃ v∥ = ∥u∥ * ∥v∥ :=
```

Comment:
I appreciate this fits your application but given the symmetry of the conclusion, I think the hypotheses should be more symmetric:
```suggestion
lemma norm_cross_product (u v : E) (h : ⟪u, v⟫ = 0) : ∥u ×₃ v∥ = ∥u∥ * ∥v∥ :=
```

Reply:
In fact, how about the converse too:
```lean
∥u ×₃ v∥ = ∥u∥ * ∥v∥ ↔ ⟪u, v⟫ = 0
```